### PR TITLE
Fix issue with optional unions when generating code

### PR DESCRIPTION
**Pull request by patrick91**

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release fixes a bug in the codegen that marked optional unions
as non optional.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @patrick91 for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: tests/codegen/snapshots/python/union.py [hunk 1]
```diff
@@ -8,5 +8,14 @@ class OperationNameResultUnionPerson:
 
 OperationNameResultUnion = Union[OperationNameResultUnionAnimal, OperationNameResultUnionPerson]
 
+class OperationNameResultOptionalUnionAnimal:
+    age: int
+
+class OperationNameResultOptionalUnionPerson:
+    name: str
+
+OperationNameResultOptionalUnion = Union[OperationNameResultOptionalUnionAnimal, OperationNameResultOptionalUnionPerson]
```
Comment (BryceBeagle) [inline tests/codegen/snapshots/python/union.py:17]: Did these files get linted?

Comment (patrick91) [reply tests/codegen/snapshots/python/union.py:17]: nope, they shouldn't be :)

Comment (BryceBeagle) [reply tests/codegen/snapshots/python/union.py:17]: Oh, codegen... 🤦

Comment (patrick91) [reply tests/codegen/snapshots/python/union.py:17]: yup, these files are snapshots, formatting them will break the tests :D

Comment (BryceBeagle) [reply tests/codegen/snapshots/python/union.py:17]: Compare ASTs 😎

**Pull request merged**